File: server/router/business/cmsArt.go
package business

import (
	v1 "go-cms/api/v1"
	"go-cms/middleware"

	"github.com/gin-gonic/gin"
)

type CmsArtRouter struct {
}

// InitCmsArtRouter 初始化 CmsArt 路由信息
func (s *CmsArtRouter) InitCmsArtRouter(Router *gin.RouterGroup) {
	router := Router.Group("cmsArt").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("cmsArt")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.CmsArtApi
	{
		router.POST("createCmsArt", apiV1.CreateCmsArt)           // 新建CmsArt
		router.POST("deleteCmsArt", apiV1.DeleteCmsArt)           // 删除CmsArt
		router.POST("deleteCmsArtByIds", apiV1.DeleteCmsArtByIds) // 批量删除CmsArt
		router.POST("updateCmsArt", apiV1.UpdateCmsArt)           // 更新CmsArt
		router.POST("quickEdit", apiV1.QuickEdit)                 // 快速编辑
	}
	{
		routerNoRecord.GET("findCmsArt", apiV1.FindCmsArt)       // 根据ID获取CmsArt
		routerNoRecord.GET("getCmsArtList", apiV1.GetCmsArtList) // 获取CmsArt列表
		routerNoRecord.GET("excelList", apiV1.ExcelList)         // 分页导出excel CmsArt列表
	}
}

File: web/src/api/cmsArt.ts
import { http,resData } from "@/utils/http";
 
// @Tags CmsArt
// @Summary 创建CmsArt
export const createCmsArt = (data?: object) => {
   return http.request<resData>("post", "/cmsArt/createCmsArt", { data });  
}

// @Tags CmsArt
// @Summary 删除CmsArt
export const deleteCmsArt = (data?: object) => {
   return http.request<resData>("post", "/cmsArt/deleteCmsArt", { data });  
}
 

// @Tags CmsArt
// @Summary 删除ByIdsCmsArt
export const deleteCmsArtByIds = (data?: object) => {
   return http.request<resData>("post", "/cmsArt/deleteCmsArtByIds", { data });  
}
 

// @Tags CmsArt
// @Summary 更新CmsArt
export const updateCmsArt = (data?: object) => {
   return http.request<resData>("post", "/cmsArt/updateCmsArt", { data });  
}

// @Tags CmsArt
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/cmsArt/quickEdit", { data });  
}
 

// @Tags CmsArt
// @Summary 用id查询CmsArt
export const findCmsArt = (params?: object) => {
   return http.request<resData>("get", "/cmsArt/findCmsArt", { params });  
}
 

// @Tags CmsArt
// @Summary 分页获取CmsArt列表
export const getCmsArtList = (params?: object) => {
   return http.request<resData>("get", "/cmsArt/getCmsArtList", { params });  
}
 

// @Tags CmsArt
// @Summary 导出excelCmsArt列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/cmsArt/excelList", { params });  
}
